src/helpers/snapshot.ts

```diff
@@ -0,0 +1,168 @@
+import { getBlockNumber } from '@snapshot-labs/snapshot.js/src/utils/web3';
+import { ipfsGet, getScores } from '@snapshot-labs/snapshot.js/src/utils';
+import { formatProposal, formatProposals } from '@/helpers/utils';
+import { getProfiles } from '@/helpers/profile';
+import getProvider from '@snapshot-labs/snapshot.js/src/utils/provider';
+import gateways from '@snapshot-labs/snapshot.js/src/gateways.json';
+import client from '@/helpers/client';
+
+const gateway = process.env.VUE_APP_IPFS_GATEWAY || gateways[0];
+
+let proposalRes;
+
+export async function getProposal(id) {
+  try {
+    console.time('getProposal.data');
+    proposalRes = await Promise.all([ipfsGet(gateway, id)]);
```

Comment:
We should get the proposal, votes and the block number in the same time here
```
 client.request(`${space.key}/proposal/${id}`),
 getBlockNumber(provider)
```
This function should return
```{ proposal, votes, blockNumber }```

These info should be passed in getResults, and getResults should return
```{ proposal, votes, results }```